### PR TITLE
Auto deploy

### PR DESCRIPTION
**Pull request by zoepage**

### CODE REVIEW

File: test/initials-test.js [hunk 1]
```diff
@@ -33,13 +38,11 @@ test('initials(namesArray)', function (t) {
 
   // https://github.com/gr2m/initials/issues/1
   t.deepEqual(initials(['j']), ['j'], 'j ☛ j')
-
-  // https://github.com/gr2m/initials/issues/14
-  t.deepEqual(initials(['Moe Minutes', 'Moe Min']), ['MoM', 'MMi'], '["Moe Minutes", "Moe Min"] ☛ ["MoM", "MMi"]')
-  t.end()
```
Comment (gr2m) [inline test/initials-test.js]: why did you remove this test?

File: initials.js [hunk 2]
```diff
@@ -169,17 +168,12 @@
...
-    for (var c = keys.length, n = 0; n < c; n++) {
-      possibleInitials = initialsForNamesMap[keys[n]]
+    for (var name in initialsForNamesMap) {
+      possibleInitials = initialsForNamesMap[name]
       optionsForNames.push(possibleInitials)
 
       for (var i = 0; i < possibleInitials.length; i++) {
-        if (duplicatesMap[possibleInitials[i]] > 0) {
-          duplicatesMap[possibleInitials[i]]--
+        if (duplicatesMap[possibleInitials[i]]) {
```
Comment (gr2m) [inline initials.js]: why the changes in initials.js when the PR is about auto-deploy?

**Comment by gr2m** [comment]
this does not yet fully finish #13, initials.js is not commonJS and there is no build task, but we can do this in two steps. But then do not move the index.html to demo/index.html, because it will break gh-pages. The index.html will be accessible at http://gr2m.github.io/initials/demo instead of http://gr2m.github.io/initials

File: .travis.yml [hunk 3]
```diff
@@ -17,3 +13,4 @@ before_script:
   - npm prune
 after_success:
   - npm run semantic-release
+  - npm run deploydocs
```
Comment (gr2m) [inline .travis.yml]: We only want to deploy docs if a release succeeded. in `after_success`, all commands get run, even if a previous one failed, I've only learned this myself recently.

We could either do `- npm run semantic-release && npm run deploydocs`, or what I would suggest is to add `"postpublish": "deploydocs"` to the package.json in `"scripts"`.

**Comment by gr2m** [comment]
The way you export initials in https://github.com/gr2m/initials/blob/auto-deploy/index.js will make this fail

```js
var initials = require('initials')
initials('John Doe')
```

Instead we'd need to do 

```js
var initials = require('initials').initials
initials('John Doe')
```

Which is not what we want. I would simply do 

```js
var exports = module.exports = require('./lib/initials')
```

Later we can then split up the `addTo`,  `find` and `parse` functions into multiple files, but let's do it in another PR :)

File: package.json [hunk 4]
```diff
@@ -3,10 +3,20 @@
   "description": "initials for names",
   "main": "initials.js",
```
Comment (gr2m) [inline package.json]: as we have `index.js` now, remove this line, otherwise `require('initials')` will fail

File: package.json [hunk 5]
```diff
@@ -3,10 +3,20 @@
...
   "scripts": {
-    "test": "standard initials.js test/initials-test.js && npm run -s test:node | tap-spec",
+    "start": "copy && beefy ./index.js --cwd=./demo",
+    "copy": "browserify lib/initials.js -s initials -o demo/initials.js",
```
Comment (gr2m) [inline package.json]: I woudl rename "copy" to "build:demo", as it's not a copy task, but a build on its own.

File: .travis.yml (hunk 3 above)
Comment (gr2m) [inline .travis.yml]: also we have to run `"build"` before release, so add `"prepublish": "build"`. And instead of "deploydocs", use "deploy", otherwise the `demo/initials.js` file will be missing

File: test/initials-test.js [hunk 6]
```diff
@@ -119,25 +141,26 @@ test('initials.parse(nameOrNames, {existing: initialsForNames})', function (t) {
...
 })
 
-// uncomment the block below to run a single test only
-// test.only('debug', function (t) {
-//   t.equal(initials('foo'), 'bar', 'debugging')
-// })
+// describe('debug', function() {
+//   equal( initials('John Doe', 3), 'JDo', 'John Doe ☛ JDo' );
+// });
```
Comment (gr2m) [inline test/initials-test.js]: Can you leave the block above with the `test.only`, instead of the current `describe` one, as this one is obsolete?

**Comment by gr2m** [comment]
Looking great, last comments, then I'll merge it and if necessary create follow up issues. Good work

**Comment by gr2m** [comment]
Looking great, last comments, then I'll merge it and if necessary create follow up issues. Good work

File: .travis.yml (hunk 3 above)
Comment (zoepage) [inline .travis.yml]: So we should change this here too:
https://github.com/hoodiehq/pouchdb-hoodie-api/blob/master/.travis.yml#L24

**Comment by gr2m** [comment]
merged via 932c043

**Comment by gr2m** [comment]
I've cleaned up your git commits, usually it should show that they have originally authored by you, and only commited by me, I'm sorry that your authorship was lost, I really didn't mean to take over your contributions :( I'll let you clean up commits in future to make sure that this won't happen again, I'm really sorry
